dist/src/main/java/io/camunda/zeebe/shared/security/SecurityConfiguration.java

```diff
@@ -39,6 +43,7 @@ public final class SecurityConfiguration {
 
   @Bean
   @ConditionalOnRestGatewayEnabled
+  @Conditional(GatewaySecurityAuthenticationEnabledCondition.class)
```

Comment:
Why don't we use that condition on the whole SecurityConfiguration class? there is no point in enabling `@EnableWebFluxSecurity` etc. if authentication disabled aka set to `none`?

Reply:
I think I got why it's here, there is also security setup for the management api in this class. But that also just disables auth completely and is thus only necessary if gateway security is configured, right? 🤔

Reply:
I will push a one liner to verify my assumption

Reply:
>I think I got why it's here, there is also security setup for the management api in this class. 

Exactly, that's why I chose not to add it to the class, but to the specific `restGatewaySecurity` method.

Moving it to the class means we disable the broker's management api when disabling the gateway security. That would be unexpected from a user's perspective I believe.

Reply:
@npepinpe our understanding is that the management API is configured to not require any authentication, can you confirm?

Reply:
It is by default, however it should still work if a user enables it, ideally.

Reply:
Not sure I follow. Let me rephrase my question.

The management API never requires authentication, regardless whether "identity-auth" is used or not it should always be publicly accessible via the dedicated management port, is that correct?

If so, it means when identity authentication is not enabled (set to none or profile not set) we could just ignore that whole SecurityConfiguration setup, right?

Reply:
What I mean is we don't expect any authentication, and it's turned off by default. But I would try to keep the behavior that if a user wants to configure some authentication, it should be possible.

Looking at the code, it seems this was refactored, and I don't know if it would be possible to define any security on the management context at all at the moment :( So I guess yes, no authentication on management context.

Reply:
>So I guess yes, no authentication on management context.

Exactly. If we want to change that we can do that in another issue/pr. For now, @megglos's fix solves the problem faced by the customer and aligns us with [the documentation](https://docs.camunda.io/docs/self-managed/zeebe-deployment/security/client-authorization/#camunda-identity-authorization).

I think we can merge this, do you both agree @npepinpe @megglos?

Reply:
I'll merge it 👍